mlir/lib/Dialect/Transform/IR/TransformInterfaces.cpp

```diff
@@ -1444,16 +1443,20 @@ bool transform::ErrorCheckingTrackingListener::failed() const {
 }
 
 void transform::ErrorCheckingTrackingListener::notifyPayloadReplacementNotFound(
-    Operation *op, ValueRange values) {
+    Operation *op, ValueRange values, ArrayRef<Operation *> aliveUsers) {
   if (status.succeeded()) {
     status = emitSilenceableFailure(
-        getTransformOp(), "tracking listener failed to find replacement op");
+        getTransformOp(), "op was replaced but replacement was of different "
+                          "kind, invalidating alive handles");
```

Comment:
Replacement may not be found for other reasons though, like replacement values belonging to different ops (so there is no _single_ replacement though all of the new producers have the right type). I suspect the message should be produced by the function that looks for replacement and forwarded here.

Reply:
You can always return additional things via references, use `std::tuple` or implement `DiagnosedSilenceableFailureOr`.